Stat svc

DmitriyDemintsev opened the pull request.

DmitriyDemintsev committed:
Выполнена реализация первой части дипломного проекта:
- реализовать сервис статистики
- реализовать HTTP-клиент для работы с сервисом статистики

DmitriyDemintsev committed:
Выполнена реализация первой части дипломного проекта:
- реализовать сервис статистики
- реализовать HTTP-клиент для работы с сервисом статистики

mikita95 commented on common-dto/pom.xml:
```diff
@@ -0,0 +1,38 @@
+        <dependency>
+            <groupId>jakarta.validation</groupId>
+            <artifactId>jakarta.validation-api</artifactId>
+            <version>2.0.2</version>
```
Номер версии лучше вынести в свойства pom.xml, так удобнее ими управлять

mikita95 commented on stats-service/src/main/java/ru/practicum/model/Stats.java:
```diff
@@ -0,0 +1,31 @@
+import javax.persistence.*;
+import java.time.LocalDateTime;
+
+@Data
```
Нужно осторожно использовать эту аннотацию, так как сейчас у вас есть lazy поля, а сейчас генерируется метод toString и hash, которые могут случайно инициализировать это поле, что приведет к лишнему вызову базы данных

mikita95 commented on stats-service/src/main/resources/schema.sql:
```diff
@@ -0,0 +1,19 @@
+  id INTEGER GENERATED BY DEFAULT AS IDENTITY NOT NULL,
+  app_id INTEGER REFERENCES applications(id),
+  uri VARCHAR NOT NULL,
+  ip VARCHAR NOT NULL,
```
IP имеет максимальный размер, можно задать ограничение

mikita95 reviewed (changes requested):
Все сделано отлично :) Оставил лишь небольшие замечания

DmitriyDemintsev committed:
учтены замечания для:
- common-dto/pom.xml
- stats-service/src/main/java/ru/practicum/model/Stats.java
- stats-service/src/main/resources/schema.sql

mikita95 reviewed (approved):
Все отлично:) Можно мержить пул-реквест
